SPEC.md

```diff
@@ -0,0 +1,172 @@
+# HTTP-WASM ABI specification
+
+> warning: This is unstable for now and can be changed in practice.
+
+## Functions implemented in the Wasm module
+
+### `_start`
+
+- params:
+  - none
+- returns:
+  - none
+
+Start function which is called when the module is loaded and initialized. This can be used by SDKs to setup and/or initialize state, but no proxy_ functions can be used at that point yet.
+
+### `proxy_on_request`
+
+- params:
+  - `i32 (uint32_t) context_id`
```

Comment:
I think this should be a uint64 as many context IDs are that length (including fasthttp)